### PR TITLE
std::weak_ptr<soci::session> to ametsuchi

### PR DESCRIPTION
**Pull request by kuvaldini**
Somehow this Gtest never exits

### CODE REVIEW

**Comment by kuvaldini** [comment]
изображение

Function blocks forever. Strange.

File: irohad/ametsuchi/impl/postgres_query_executor.hpp [hunk 1]
```diff
@@ -38,11 +38,16 @@ namespace iroha {
       bool validate(const shared_model::interface::BlocksQuery &query,
                     const bool validate_signatories) override;
 
+      //     private:
+      //      std::shared_ptr<soci::session> sql(){
+      //        return std::shared_ptr<soci::session>(sql_);
+      //      }
+
```
Comment (kuvaldini) [inline irohad/ametsuchi/impl/postgres_query_executor.hpp:45]:
```suggestion
```

File: test/module/irohad/ametsuchi/ametsuchi_test.cpp [hunk 2]
```diff
@@ -780,6 +783,9 @@ TEST_F(PreparedBlockTest, PrepareBlockNoStateChanged) {
  * @then state of the ledger is changed
  */
 TEST_F(PreparedBlockTest, CommitPreparedStateChanged) {
+  if (not storage->preparedBlocksEnabled())
+    GTEST_SKIP();
+
```
Comment (kuvaldini) [inline test/module/irohad/ametsuchi/ametsuchi_test.cpp:788]:
```suggestion
```

File: irohad/ametsuchi/mutable_factory.hpp [hunk 3]
```diff
@@ -49,6 +49,9 @@ namespace iroha {
       /// Check if prepared commits are enabled.
       virtual bool preparedCommitEnabled() const = 0;
 
+      /// Check if prepared blocks are enabled.
+      virtual bool preparedBlocksEnabled() const = 0;
```
Comment (lebdron) [inline irohad/ametsuchi/mutable_factory.hpp:53]: Can be removed

Comment (kuvaldini) [reply irohad/ametsuchi/mutable_factory.hpp:53]: Will not compile

Comment (lebdron) [reply irohad/ametsuchi/mutable_factory.hpp:53]: Have to remove all the usages

File: irohad/ametsuchi/impl/mutable_storage_impl.cpp [hunk 4]
```diff
@@ -44,7 +46,7 @@ namespace iroha {
           block_storage_(std::move(block_storage)),
           committed(false),
           log_(log_manager->getLogger()) {
-      sql_ << "BEGIN";
+      *sql() << "BEGIN";
```
Comment (kuvaldini) [inline irohad/ametsuchi/impl/mutable_storage_impl.cpp:49]:
```
assert(command_executor->getSession());
*command_executor->getSession() << "BEGIN";
sql_ = command_executor->getSession();
```

Comment (kuvaldini) [reply irohad/ametsuchi/impl/mutable_storage_impl.cpp]: fixed

**Comment by lebdron** [comment]
@kuvaldini tests don't seem to pass now, can you check locally and on the build server again?

**Pull request closed without merging**